### ISSUE TITLE
[Fix Available 1.308] Weld edges issue

### ISSUE THREAD

**Issue by Hisashimaru** [LogicalError/realtime-CSG-for-unity#30]
Unity console outputs errors, and Realtime CSG stuck on the surface mode after weld edges.
![gif](https://cloud.githubusercontent.com/assets/7605365/19423041/442e8d0e-9457-11e6-8190-976f6e1d3e51.gif)

Realtime CSG 1.305
Unity 5.4.2f1

**Comment by LogicalError** [comment]
Ok, that's weird. I've never seen that before .. does this always happen for you? Or just with that particular brush? If you can reproduce this, what exactly is the full error message?
Thanks

**Comment by Hisashimaru** [comment]
The problem always happen for me..

Unity outputs this error logs.


```
IndexOutOfRangeException: Array index is out of range.
RealtimeCSG.SurfaceEditBrushTool.DrawOutlines () (at Assets/Plugins/RealtimeCSG/Editor/Scripts/EditorWindows/BrushTools/SurfaceEditBrushTool.cs:677)
RealtimeCSG.SurfaceEditBrushTool.HandleEvents () (at Assets/Plugins/RealtimeCSG/Editor/Scripts/EditorWindows/BrushTools/SurfaceEditBrushTool.cs:1001)
RealtimeCSG.CSGBrushEditorManager.OnSceneGUI (UnityEditor.SceneView sceneView) (at Assets/Plugins/RealtimeCSG/Editor/Scripts/EditorWindows/BrushEditorManager/CSGBrushEditorManager.cs:631)
RealtimeCSG.UnitySceneEventHandler.OnScene (UnityEditor.SceneView sceneView) (at Assets/Plugins/RealtimeCSG/Editor/Scripts/Managers/UnitySceneEventHandler.cs:880)
UnityEditor.SceneView.CallOnSceneGUI () (at C:/buildslave/unity/build/Editor/Mono/SceneView/SceneView.cs:2046)
UnityEditor.SceneView.HandleSelectionAndOnSceneGUI () (at C:/buildslave/unity/build/Editor/Mono/SceneView/SceneView.cs:1381)
UnityEditor.SceneView.OnGUI () (at C:/buildslave/unity/build/Editor/Mono/SceneView/SceneView.cs:1218)
System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.Reflection/MonoMethod.cs:222)
```

**Comment by LogicalError** [comment]
Allright I'll look into this, thanks

**Comment by LogicalError** [comment]
Fixed in version 1.306, which is available on the update site ( https://logicalerror.github.io/ )
Once this version is on the asset store I'll close this bug.

**Comment by Hisashimaru** [comment]
Thanks!
I confirmed the fix, but I can't select some surfaces.
![surfaces](https://cloud.githubusercontent.com/assets/7605365/19465156/ba5c0db8-953c-11e6-8c9a-ffb0f0281c39.gif)

**Comment by LogicalError** [comment]
Ah sorry, I didn't notice that when I fixed the other bug. Unfortunately I'm away this week and won't be back until next Tuesday. A workaround would be to use clipping instead until I fix this. Sorry for the inconvenience

**Comment by LogicalError** [comment]
This is fixed in version 1.308 which is available on the update site ( http://prenominal.nl/updates/index.php )!
I will close this ticket when it's available on the asset store

**Comment by LogicalError** [comment]
Fix is in the version that's on the asset store now, so closing ticket
